### PR TITLE
[backport] Do not ignore FutureWarning other than experimental features

### PR DESCRIPTION
**Pull request by kmaehashi**
Backport of #5949

(conflict resolved manually)

### CODE REVIEW

**Comment by toslunar** [comment]
How about `cherry-pick`ing cc6f54cb226e28fbcd87f8f5a1b3f318bee99fd8 here?

**Comment by kmaehashi** [comment]
Jenkins, test this please.

File: tests/chainer_tests/functions_tests/array_tests/test_split_axis.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import unittest
+import warnings
```
Comment (toslunar) [inline tests/chainer_tests/functions_tests/array_tests/test_split_axis.py]: It seems that no changes are needed in `tests/chainer_tests/functions_tests/array_tests/test_split_axis.py`.

Comment (kmaehashi) [reply tests/chainer_tests/functions_tests/array_tests/test_split_axis.py]: Thanks, resolved.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**